### ISSUE TITLE
Configuration wasm32-unknown-wasi not supported 

### ISSUE THREAD

**Issue by sbellem** [oxalica/rust-overlay#75]
I'm trying to follow the [cross compilation example for wasm32-wasi](https://github.com/oxalica/rust-overlay/tree/master/examples/cross-wasi), and get the following when running `nix-shell`:

```shell
*** Configuration wasm32-unknown-wasi not supported
make[1]: *** [Makefile:4332: configure-gcc] Error 1
make[1]: Leaving directory '/build/build'
make: *** [Makefile:956: all] Error 2
error: builder for '/nix/store/3q38zwcxbzkz8l3xa390fg21s1c6qw78-wasm32-wasi-stage-final-gcc-debug-10.3.0.drv' failed with exit code 2;
       last 10 log lines:
       > checking whether times is declared... yes
       > checking whether sigaltstack is declared... yes
       > checking for struct tms... yes
       > checking for clock_t... yes
       > checking for F_SETLKW... yes
       > checking if mkdir takes one argument... no
       > *** Configuration wasm32-unknown-wasi not supported
       > make[1]: *** [Makefile:4332: configure-gcc] Error 1
       > make[1]: Leaving directory '/build/build'
       > make: *** [Makefile:956: all] Error 2
       For full logs, run 'nix log /nix/store/3q38zwcxbzkz8l3xa390fg21s1c6qw78-wasm32-wasi-stage-final-gcc-debug-10.3.0.drv'.
error: 1 dependencies of derivation '/nix/store/jl9q2c2iih8gvw6l5ccw2gww97bqxsvi-rustc-1.58.1-x86_64-unknown-linux-gnu.drv' failed to build
error: 1 dependencies of derivation '/nix/store/jxkalmd4b3j5vr1lcj0nvawj6qwggdvk-rust-minimal-1.58.1.drv' failed to build
error: build of '/nix/store/5zql8f3mnv53qss2v6w9dwbh5ajrrgrj-stdenv-linux.drv', '/nix/store/jxkalmd4b3j5vr1lcj0nvawj6qwggdvk-rust-minimal-1.58.1.drv' failed
```

I've tried a few things, such as:

* changing the `nixpkgs` import, to the commit https://github.com/oxalica/nixpkgs/commit/d96fe72ef5c93419f99f6d78b2dc6c1d54c8bd23
* change the `config` to `"wasm32-unknown-wasi"` (instead of `wasm32-wasi`)
* use `nixpkgs-unstable`
* use a rust nightly toolchain

**Comment by oxalica** [comment]
Seems caused by the build failure of `gcc` from `nixpkgs`. IIRC, `clang` used to be the default C compiler for WASI (and is used as the linker by `rustc`) but now it's `gcc`, while `gcc` doesn't support WASI. I'll look into it later if time permits.

Maybe you can try overriding `stdenv` with `clangStdenv`.

**Comment by sbellem** [comment]
Ok thanks, that is helpful, I will try to understand better how that works.

**Comment by oxalica** [comment]
Fixed in 44801306a2aa0e9aaa47588d615ce6df4acf18c6. The dependency to CC from nixpkgs is removed since `rustc` can do it by itself.
